filesource.go

```diff
@@ -216,48 +216,53 @@ func (s *FileSource) runMergeFile() error {
 	}
 }
 
-func (s *FileSource) streamIncomingFile(newIncomingFile *incomingBlocksFile, blocksStore dstore.Store) error {
-	defer func() {
-		close(newIncomingFile.blocks)
-	}()
-
-	atomic.AddInt64(&currentOpenFiles, 1)
-	s.logger.Debug("open files", zap.Int64("count", atomic.LoadInt64(&currentOpenFiles)), zap.String("filename", newIncomingFile.filename))
-	defer atomic.AddInt64(&currentOpenFiles, -1)
-
-	// FIXME: Eventually, RETRY for this given file.. and continue to write to `newIncomingFile`.
-	reader, err := blocksStore.OpenObject(context.Background(), newIncomingFile.filename)
-	if err != nil {
-		return fmt.Errorf("fetching %s from block store: %w", newIncomingFile.filename, err)
-	}
-	defer reader.Close()
+type retryableError struct {
```

Comment:
No need for all this, a simple `type retryableError error` can be used. Even `isRetryable` could be removed, a direct

`if _, ok := err.(retryableError); ok { ... }` feels as good as `isRetryable` IMO.

Reply:
Only thing left is thing change IMO